scripts/scil_remove_similar_streamlines.py

```diff
@@ -0,0 +1,169 @@
+    min_distance = args[1]
+    cluster_thr = args[2]
+    min_cluster_size = args[3]
+    average_streamlines = args[4]
+
+    min_cluster_size = max(min_cluster_size, 1)
+    thresholds = [40, 30, 20, cluster_thr]
+    cluster_map = qbx_and_merge(ArraySequence(streamlines),
+                                thresholds,
+                                nb_pts=20,
+                                verbose=False)
+
+    return subsample_clusters(cluster_map, streamlines, min_distance,
+                              min_cluster_size, average_streamlines)
+
+
+def _buildArgsParser():
+    p = argparse.ArgumentParser(formatter_class=argparse.RawTextHelpFormatter,
+                                description=__doc__)
+
+    p.add_argument('in_bundle',
+                   help='Path of the input bundle.')
+
+    p.add_argument('min_distance', type=float,
+                   help='Distance threshold for 2 streamlines to be '
+                        'considered similar (mm).')
+
+    p.add_argument('out_bundle',
+                   help='Path of the output tractography file')
+
+    p.add_argument('--clustering_thr', type=float, default=6,
+                   help='Clustering threshold for QB/QBx (mm), during '
+                        'the first approximation [%(default)s].')
+    p.add_argument('--min_cluster_size', type=int, default=5,
+                   help='Minimum cluster size for the first iteration '
+                        '[%(default)s].')
+    p.add_argument('--convergence', type=int, default=100,
+                   help='Streamlines count difference threshold to stop '
+                        're-running the algorithm [%(default)s].')
+    p.add_argument('--avg_similar', action='store_true',
+                   help='Average similar streamlines rather than removing them'
+                        '[%(default)s]. Requires a small min_distance.')
+    p.add_argument('--processes', type=int, default=1,
+                   help='Number of desired processes [%(default)s].')
+
+    add_reference_arg(p)
+    add_overwrite_arg(p)
+    add_verbose_arg(p)
+
+    return p
+
+
+def main():
+    parser = _buildArgsParser()
+    args = parser.parse_args()
+
+    assert_inputs_exist(parser, args.in_bundle)
+    assert_outputs_exist(parser, args, args.out_bundle)
+
+    log_level = logging.WARNING
+    if args.verbose:
+        log_level = logging.DEBUG
+    logging.basicConfig(level=log_level)
+
+    sft = load_tractogram_with_reference(parser, args, args.in_bundle)
+    streamlines = list(sft.streamlines)
+    original_length = len(streamlines)
+    logging.debug('Loaded %s streamlines...', original_length)
+
+    pool = multiprocessing.Pool(args.processes)
+    timer = time()
+
+    logging.debug('Lauching subsampling on %s processes.', args.processes)
+    last_iteration = False
+    while True:
+        if len(streamlines) < 1000:
+            logging.warning('Subsampling less than 1000 streamlines is risky.')
+            break
```

Comment:
Can you make this robust to number of iterations? Should not say this if original bundle has more than 1000 streamlines